09_json/runnerdave/cmd/puppy_server/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	puppy "github.com/anz-bank/go-course/09_json/runnerdave/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/runnerdave/pkg/puppy/store"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	args           = os.Args[1:]
+	data           = kingpin.Flag("data", "data file").Short('d').Default("data.json").ExistingFile()
+	out  io.Writer = os.Stdout
+)
+
+func main() {
+	_, err := kingpin.CommandLine.Parse(args)
+	if err != nil {
+		fmt.Fprintf(out, "Command line could not be parsed, error: %v", err)
+		return
+	}
+
+	d, _ := ioutil.ReadFile(*data)
+	puppies, uerr := unmarshalPuppies(d)
+	if uerr != nil {
+		return
+	}
+
+	mapStore := storeInMap(puppies)
+	fmt.Fprintf(out, "Map store of puppies:%v", mapStore)
+
+	syncStore := storeInSync(puppies)
+	fmt.Fprintf(out, "Sync store of puppies:%v", syncStore)
+}
+
+func unmarshalPuppies(d []byte) ([]puppy.Puppy, error) {
+	puppies := make([]puppy.Puppy, 0)
```

Comment:
This is up to personal preference, but I try to avoid `make` where I can (looks noisy to me)
You can initialize with 
```
puppies := []puppy.Puppy{}
```

Note that if you want to initalize with a non-zero capacity, you have to use `make`, so `make` cannot always be avoided.

Reply:
agree, much nicer this way